crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods/string_methods.rs

```diff
@@ -0,0 +1,90 @@
+#[expect(clippy::cast_possible_wrap, clippy::cast_possible_truncation, clippy::cast_sign_loss)]
+pub(super) fn evaluate_string_index_of<'a>(
+    string: &'a str,
+    search_value: Option<&'a str>,
+    from_index: Option<f64>,
+) -> isize {
+    let from_index = from_index.map_or(0.0, |x| x.floor().max(0.0)) as usize;
+
+    let Some(search_value) = search_value else {
+        return -1;
+    };
+    let result = string.chars().skip(from_index).collect::<String>().find(search_value);
+    result.map(|index| index + from_index).map_or(-1, |index| index as isize)
+}
+
+#[expect(clippy::cast_possible_wrap, clippy::cast_possible_truncation, clippy::cast_sign_loss)]
+pub(super) fn evaluate_string_last_index_of<'a>(
+    string: &'a str,
+    search_value: Option<&'a str>,
```

Comment:
Can be `&'a str` instead of Option

Reply:
if we change it to that instead of option, it makes it harder to handle the case `"foo".indexOf()` where a search string it not defined

Reply:
Get it, good.